src/test/java/de/tum/in/www1/artemis/programmingexercise/ProgrammingExerciseResultBambooIntegrationTest.java

```diff
@@ -38,27 +40,42 @@ void tearDown() {
     @Test
     @WithMockUser(value = "student1", roles = "USER")
     public void shouldUpdateTestCasesAndResultScoreFromSolutionParticipationResult() {
+        programmingExerciseResultTestService.setup();
         var notification = ModelFactory.generateBambooBuildResult(Constants.ASSIGNMENT_REPO_NAME, List.of("test1", "test2", "test4"), List.of());
         programmingExerciseResultTestService.shouldUpdateTestCasesAndResultScoreFromSolutionParticipationResult(notification);
     }
 
-    @Test
+    @ParameterizedTest
+    @EnumSource(value = ProgrammingLanguage.class, names = { "JAVA", "SWIFT" })
     @WithMockUser(value = "student1", roles = "USER")
-    public void shouldStoreFeedbackForResultWithStaticCodeAnalysisReport() {
-        var notification = ModelFactory.generateBambooBuildResultWithStaticCodeAnalysisReport(Constants.ASSIGNMENT_REPO_NAME, List.of("test1"), List.of());
-        programmingExerciseResultTestService.shouldStoreFeedbackForResultWithStaticCodeAnalysisReport(notification);
+    public void shouldStoreFeedbackForResultWithStaticCodeAnalysisReport(ProgrammingLanguage programmingLanguage) {
+        programmingExerciseResultTestService.setupForProgrammingLanguage(programmingLanguage);
+        var notification = ModelFactory.generateBambooBuildResultWithStaticCodeAnalysisReport(Constants.ASSIGNMENT_REPO_NAME, List.of("test1"), List.of(), programmingLanguage);
+        var scaReports = notification.getBuild().getJobs().get(0).getStaticCodeAnalysisReports();
+        if (programmingLanguage == ProgrammingLanguage.SWIFT) {
+            // SwiftLint has only one category at the moment
+            assertThat(scaReports.size()).isEqualTo(1);
+            assertThat(scaReports.get(0).getIssues().get(0).getCategory()).isEqualTo("swiftLint");
+        }
+        else if (programmingLanguage == ProgrammingLanguage.JAVA) {
+            assertThat(scaReports.size()).isEqualTo(4);
+            scaReports.get(0).getIssues().forEach(issue -> assertThat(issue.getCategory()).isNotNull());
+        }
+        programmingExerciseResultTestService.shouldStoreFeedbackForResultWithStaticCodeAnalysisReport(notification, programmingLanguage);
     }
 
     @Test
     @WithMockUser(value = "student1", roles = "USER")
     public void shouldStoreBuildLogsForSubmission() {
+        programmingExerciseResultTestService.setup();
```

Comment:
why do you move the setup method to here?
The setup method should be called in the setup method above and not in each test case!

Reply:
the setup method always creates a Java exercise, so we would need a parameterized @BeforeEach method, which always initializes the correct programming language exercise.   
Therefore I needed to split the setup() method into `setup()` and `setupForProgrammingLanguage(programmingLanguage)`. 

Otherwise, each time the testcase is non-Java, we would create an exercise with Java and an exercise for the specified prog. language.

Reply:
resolved in ba1458f587a5d5b3c9c465d0ab19e1c675d5d3f0